### PR TITLE
Bugfix error handling misses some important edge cases

### PR DESCRIPTION
**Pull request by G-Salviati**
This PR:

- fixes routes handlers not working in case of exceptions in the process;
- adds tests for some of db failures during requests;
- adds a logic for which only a MongoDB instance can exist (kinda a singleton).

### CODE REVIEW

File: server/app.js [hunk 1]
```diff
@@ -3,8 +3,13 @@ var path = require("path");
 var cookieParser = require("cookie-parser");
 var logger = require("morgan");
 var cors = require("cors");
+var dao = require("./dao/dao");
 
+const {
+  checkValidationErrorMiddleware,
+} = require("./handlers/shared_validators");
 var employeeHandlers = require("./handlers/employee");
+const { domainToASCII } = require("url");
```
Comment (MarcoBramini) [inline server/app.js]: wtf is that

**Review by MarcoBramini — CHANGES_REQUESTED** [review]

**Review by MarcoBramini — APPROVED** [review]
